forem/forem
# Fix erroring data update script

## What type of PR is this? (check all applicable)


## Description

After deploying the Forem fleet, we saw a flood of errors come through from a data update script that was erroring out because of a `NoMethodError` (`undefined method users for nil:NilClass`). The script did not account for Forems where `Role.find_by(name: "tech_admin")` did not return anything, which is what caused the error.

## Related Tickets & Documents

Relevant [Honeybadger errors across Forem fleet](https://app.honeybadger.io/search?q=NoMethodError%3A%E2%80%81undefined+method%E2%80%81%60users%27+for+nil%3ANilClass)

## QA Instructions, Screenshots, Recordings

Please pull down this branch and ensure you have _no users_ with the `:tech_admin` role. Run `bin/setup` which should invoke `rake data_updates:run`. You should _not_ see any errors!

### UI accessibility concerns?

N/A

## Added tests?

      have not been included_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] [Storybook](https://storybook.forem.com/) (for Crayons components)

## Are there any post deployment tasks we need to perform?

**YES**: we need to make sure this script executes on DEV, and we need to redeploy the Forem fleet and ensure that this script runs properly.

If all of that ^^ happens as expected, we can resolve this error across Honeybadger.
